### PR TITLE
Fight page

### PR DESCRIPTION
**Pull request by MehdiDAAJI**

### CODE REVIEW

**Review by serhanif59 — APPROVED** [review]

File: public/assets/css/battle.css [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    width: 20%;
+    padding: 15px 60px;
+    border-radius: 60px;
+    background: linear-gradient(to bottom, var(--primary-color) 5%, var(--primary-hover) 100%);
+    text-align: center;
+}
+
+/*
+
+main.fight section.containerRedButton {
```
Comment (Benja-san) [inline public/assets/css/battle.css]: clean dead code

File: src/Controller/FightController.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+namespace App\Controller;
+
+use App\Model\AttackManager;
+use App\Model\UnicornManager;
+
+class FightController extends AbstractController
+{
+    /**
+     * Display about page
```
Comment (Benja-san) [inline src/Controller/FightController.php]: display fight page

File: src/Controller/FightController.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+     */
+    public function index(): string
+    {
+        $unicornManager = new AttackManager();
```
Comment (Benja-san) [inline src/Controller/FightController.php]: coherence between variables and Object, please change unicornManager for attackManager

File: src/Controller/FightController.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    public function confirmOpponentAttack(): void
+    {
+        $_SESSION["opponentAttackId"] = $_GET["id"];
+    }
+
+    public function confirmAttack(): string|null
+    {
+
+        if ($_GET) {
+            if ($_GET['type'] == 'user') {
```
Comment (Benja-san) [inline src/Controller/FightController.php]: triple = required

File: src/Controller/FightController.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                $attacks = $this->confirmUserAttack();
+                return $this->twig->render('Fight/selectAttack.html.twig', [
+                    "type" => "opponent",
```
Comment (Benja-san) [inline src/Controller/FightController.php:40]: why using a type key when you have an opponent array in there

File: src/View/Fight/selectAttack.html.twig [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+<main class="fight">
+
+    <section class="battle">
+        <img class="player" src="/assets/images/profile/{{sessionUser ? sessionUser.avatar : 'img_avatar.png'}}">
+        <img class="versus" src="/assets/images/vs-index.png">
+        <img class="player" src="/assets/images/profile/{{opponentUser ? opponentUser.avatar : 'img_avatar.png'}}">
+    </section>
+
+    <section class="round">
+        <h1> ROUND 1 </h1>
```
Comment (Benja-san) [inline src/View/Fight/selectAttack.html.twig]: should be incremented dynamically, if you don't hae time to do it, remove the number to only let "Round" as a title

File: src/View/Fight/selectAttack.html.twig [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+ </div>
+  
+    <section class="round">
+        <h2> utilise l'une des capacités de ta licorne !</h2>
+    </section>
+
+    <section class="containerRedButton">
+    {% for attack in attacks %}                        
+        <div>
+            <a href="confirmAttack?type={{type}}&&id={{attack.attId}}" class="selectAttack">{{attack.attName}} </a>
```
Comment (Benja-san) [inline src/View/Fight/selectAttack.html.twig]: one & needed

**Review by Benja-san — CHANGES_REQUESTED** [review]

File: src/View/Fight/selectAttack.html.twig (hunk 6 above)
Comment (MehdiDAAJI) [reply src/View/Fight/selectAttack.html.twig]: i change for ROUND, but it is ali who will feed the round once the branch merges

File: src/Controller/FightController.php (hunk 5 above)
Comment (MehdiDAAJI) [reply src/Controller/FightController.php:40]: to know if it is the user or the opponent round

**Review by Benja-san — APPROVED** [review]

**Pull request merged**